Detectors/Upgrades/ALICE3/Passive/src/Pipe.cxx

```diff
@@ -93,75 +94,140 @@ void Alice3Pipe::ConstructGeometry()
-  TGeoCompositeShape* outerBerylliumTubeVacuumComposite;                                                                                                   // Composite volume to subctract to vacuum
-  TGeoVolume* outerBerylliumTubeVacuumVolume;                                                                                                              // Final volume to be used
+  // We split the naming of the parts if the beam pipe for ALICE 3 into parts
+  // - pipe
+  // - vacuum vessel (which hosts the primary vacuum)
+  // - iris vacuum vessel (which hosts the secondary vacuum)
 
-  TString compositeFormula{"OUT_PIPEVACUUM_BASEsh"}; // If pipe is alone we won't subctract anything
+  // A3IP update
+  // Vacuum
+  TGeoTube* vacuumBasePipe = new TGeoTube("PIPEVACUUM_BASEsh", 0., mPipeRIn, mA3IPLength / 2.);
+  TGeoTube* vacuumBaseVacuumVessel = new TGeoTube("VACUUM_VESSELVACUUM_BASEsh", mPipeRIn, mVacuumVesselRIn, mVacuumVesselLength / 2.);
+  // Excavate volumes from the vacuum such that there is place for the TRK barrel layers and FT3 disc layers of the IRIS tracker
+  // And the other passive shapes: coldplate, iris tracker vacuum vessel
+  TGeoCompositeShape* vacuumComposite;
+  TGeoVolume* vacuumVolume;
+  TString compositeFormula{"PIPEVACUUM_BASEsh+VACUUM_VESSELVACUUM_BASEsh"};
   TString subtractorsFormula;
 
   if (!mIsTRKActivated) {
     std::vector<TGeoTube*> trkLayerShapes;
 
-    std::vector<std::array<float, 3>> layersQuotas = {std::array<float, 3>{0.5f, 50.f, 50.e-4}, // TODO: Set layers dynamically. {radius, zLen, thickness}
-                                                      std::array<float, 3>{1.2f, 50.f, 50.e-4},
-                                                      std::array<float, 3>{2.5f, 50.f, 50.e-4}};
+    std::vector<std::array<float, 3>> layersQuotas = {std::array<float, 3>{0.5f, 50.f, 100.e-4}, // TODO: Set layers dynamically. {radius, zLen, thickness}
+                                                      std::array<float, 3>{1.2f, 50.f, 100.e-4},
+                                                      std::array<float, 3>{2.5f, 50.f, 100.e-4}};
 
-    subtractorsFormula = "TRKLAYER_0sh";                           // First volume to be subctracted (at least one has to be provided)
+    subtractorsFormula = "";                                       // First volume to be subctracted (at least one has to be provided)
     for (auto iLayer{0}; iLayer < layersQuotas.size(); ++iLayer) { // Create TRK layers shapes
       auto& layerData = layersQuotas[iLayer];
-      trkLayerShapes.emplace_back(new TGeoTube(Form("TRKLAYER_%dsh", iLayer), layerData[0], layerData[0] + layerData[2], layerData[1] / 2));
-      if (iLayer > 0) {
-        subtractorsFormula += Form("+TRKLAYER_%dsh", iLayer);
+      trkLayerShapes.emplace_back(new TGeoTube(Form("TRKLAYER_%dsh", iLayer), layerData[0], layerData[0] + layerData[2], layerData[1] / 2.));
+      if (iLayer != 0) {
+        subtractorsFormula += Form("+");
```

Comment:
same here:
```suggestion
      if (!iLayer) {
        subtractorsFormula += "+";
```

Reply:
I removed "Form", but iLayer != 0 should stay for this logic